### PR TITLE
Server wms bbox wrong transform on empty layers

### LINKED ISSUE #30264: WMS wrong layer bbox when layer is empty and CRS is different than the project
Probably missing a coord transform in 

qgswmsgetcapabilities.cpp

```
                // if there's no feature, use the wms extent defined in the
                // project...
                extent = QgsServerProjectUtils::wmsExtent( *project );
                if ( extent.isNull() )
                {
                  // or the CRS extent otherwise
                  extent = vl->crs().bounds();
                }
```

### PR DESCRIPTION
**Pull request by elpaso**
... and different CRS

Fixes #30264

### CODE REVIEW

File: src/server/services/wms/qgswmsgetcapabilities.cpp [hunk 1]
```diff
@@ -1073,6 +1073,12 @@ namespace QgsWms
                   // or the CRS extent otherwise
                   extent = vl->crs().bounds();
                 }
+                // If CRS is different transform it to layer's CRS
+                else if ( vl->crs() != project->crs() )
+                {
+                  QgsCoordinateTransform ct( project->crs(), vl->crs(), project->transformContext() );
+                  extent = ct.transform( extent );
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetcapabilities.cpp]: Do you need to catch cs exceptions here?

Comment (elpaso) [reply src/server/services/wms/qgswmsgetcapabilities.cpp]: yes, but I'm in doubt about what to do in that case: abort and return an exception to the client?

Comment (elpaso) [reply src/server/services/wms/qgswmsgetcapabilities.cpp]: ... or just log and skip the offending layer?

Comment (pblottiere) [reply src/server/services/wms/qgswmsgetcapabilities.cpp]: +1 for logging, I think you did well!

**Comment by pblottiere** [comment]
It look good to me, thanks @elpaso :+1:!

**Review by pblottiere — APPROVED** [review]

**Review by rldhont — APPROVED** [review]

**Comment by elpaso** [comment]
@rldhont , @pblottiere thank you for the review!

**Pull request merged**